neuraxle/metaopt/trial.py

```diff
@@ -50,32 +55,224 @@ def __init__(
             self,
             hyperparams: HyperparameterSamples,
             status: 'TRIAL_STATUS' = None,
+            pipeline: BaseStep = None,
+            validation_splits=None,
+            cache_folder=None,
+            error=None,
+            error_traceback=None,
+            start_time=None,
+            end_time=None
+    ):
+        if status is None:
+            status = TRIAL_STATUS.PLANNED
+        if validation_splits is None:
+            validation_splits = []
+
+        self.status: TRIAL_STATUS = status
+        self.hyperparams = hyperparams
+        self.pipeline: BaseStep = pipeline
+        self.validation_splits = validation_splits
+        self.cache_folder = cache_folder
+        self.error_traceback = error_traceback
+        self.error = error
+        self.start_time = start_time
+        self.end_time = end_time
+
+    def new_validation_split(self, pipeline: BaseStep):
+        """
+        Create a new trial split.
+        A trial has one split when the validation splitter function is validation split.
+        A trial has one or many split when the validation splitter function is kfold_cross_validation_split.
+
+        :type pipeline: pipeline to execute
+        :return:
+        """
+        trial_split: TrialSplit = TrialSplit(split_number=len(self.validation_splits))
+        self.validation_splits.append(trial_split)
+
+        return trial_split.set_pipeline(pipeline)
+
+    def set_pipeline(self, pipeline: BaseStep):
+        """
+        Set pipeline.
+
+        :param pipeline: fitted pipeline
+        :return:
+        """
+        self.pipeline = pipeline
+
+    def save_model(self):
+        """
+        Save fitted model in the trial hash folder.
+        """
+        hyperparams = self.hyperparams.to_flat_as_dict_primitive()
+        trial_hash = self._get_trial_hash(hyperparams)
+        self.pipeline.set_name(trial_hash).save(ExecutionContext(self.cache_folder), full_dump=True)
+
+    def set_hyperparams(self, hyperparams: HyperparameterSamples):
+        """
+        Set trial hyperparams.
+
+        :param hyperparams: trial hyperparams
+        :return:
+        """
+        self.hyperparams = hyperparams
+
+    def get_higher_score_is_better(self):
```

Comment:
`is_higher_score_better()` ?

Reply:
returns bool type

Reply:
no it really gets if an higher score would be better for the main metric